pkg/scheduler/framework/runtime/framework.go

```diff
@@ -686,10 +696,14 @@ func (f *frameworkImpl) RunPreFilterPlugins(ctx context.Context, state *framewor
 			if len(pluginsWithNodes) == 1 {
 				msg = fmt.Sprintf("node(s) didn't satisfy plugin %v", pluginsWithNodes[0])
 			}
-			return nil, framework.NewStatus(framework.Unschedulable, msg)
+			// In this case, the preemption should happen later in this scheduling cycle.
+			// So we need to execute all PreFilter.
+			// https://github.com/kubernetes/kubernetes/issues/119770
+			returnStatus = framework.NewStatus(framework.Unschedulable, msg)
```

Comment:
I think we should consider the Nodes that aren't returned in the PreFilterResult as UnschedulableAndUnresolvable (I think that applies to the existing plugins that implement it). Or let the plugins give a result.

Reply:
Good point. This would benefit the preemption performance of the combination of a plugin that returns non-nil PreFilterResult (e.g., ResourceFit for deamonset pod) + a plugin that return Unschedulable. So we can narrow down the search scope of preemption candidates to the union of PreFilterResult(s); instead of going through all nodes for preemption.